tests/collection/from_directory.py

```diff
@@ -0,0 +1,27 @@
+#! /usr/bin/env python3
+
+import os
+import WrightTools as wt
+from WrightTools import datasets
+
+def test_from_directory():
+    from_dict = {'*.data':wt.data.from_PyCMDS,
+                 '*.csv':wt.collection.from_Cary,
+                 'KENT':None,
+                 'COLORS':None,
+                }
+    p = wt.datasets.PyCMDS.w1_000
+    p = os.path.dirname(p)
+    p = os.path.dirname(p)
+    col = wt.collection.from_directory(p, from_dict)
+    # Print the tree in case tests fail, also will catch errors in printing
+    col.print_tree()
+    assert col.natural_name == 'datasets'
+    assert col.PyCMDS['w1 000']
```

Comment:
are we getting invalid identifiers as names? seems reasonable to coerce to identifier, since natural naming is so core to the "WrightTools experience"

I consider this a bug right now, but am happy to be convinced otherwise

Reply:
My goal was to replicate the file structure as closely as possible, without messing with names if I could avoid it. I personally tend to have file names which are valid identifiers (without the extension, that is). I do this in part so this is not an issue in the things I do.

I'll note, prior behavior, without passing the name to the from method, does not guarantee a python identifier either: I have datasets which have the automatically generated name start with a numeral.

Reply:
To kind of sum up my thoughts: I don't think we should dictate what the names can or can't be. Having natural namespace is, to me, a useful thing that I wish to utilize, but it is a nicety, not a requirement. I do not want to build in the assumption that all file names will be valid python identifiers, in part because I do not want to lock the format specification to python